### PR TITLE
Fix net peer attribute for unix socket connection

### PR DESCRIPTION
**Pull request by shadchin**
# Description

After switch on `.get(...)`, `KeyError` exception is never raised

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

```
import redis
from opentelemetry.instrumentation.redis import RedisInstrumentor

RedisInstrumentor().instrument()

client = redis.Redis(unix_socket_path="/foobar")  # or redis.Redis.from_url("unix:///foobar")
# Span attribute `net.peer.name` must be equal to `/foobar`
```

# Does This PR Require a Core Repo Change?

- [ ] Yes. - Link to PR: 
- [x] No.

# Checklist:

See [contributing.md](https://github.com/open-telemetry/opentelemetry-python-contrib/blob/main/CONTRIBUTING.md) for styleguide, changelog guidelines, and more.

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

### CODE REVIEW

**Review by xrmx — COMMENTED** [review]
Thanks for the PR, please add a test so we can avoid regressions

File: instrumentation/opentelemetry-instrumentation-redis/tests/test_redis.py [hunk 1]
```diff
@@ -226,3 +231,83 @@ def test_no_op_tracer_provider(self):
...
+            span.attributes[SpanAttributes.NET_TRANSPORT],
+            NetTransportValues.IP_TCP.value,
+        )
+
+    def test_attributes_unix_socket(self):
+        redis_client = redis.Redis.from_url(
+            "unix://foo@/path/to/socket.sock?db=3&password=bar"
+        )
+
+        with mock.patch.object(redis_client, "connection"):
```
Comment (xrmx) [inline instrumentation/opentelemetry-instrumentation-redis/tests/test_redis.py:292]: What do you think on adding an `assert_called_once_with` on this mock so we can check the connection parameters?

**Review by xrmx — APPROVED** [review]
Thanks a lot for adding tests!

File: instrumentation/opentelemetry-instrumentation-redis/tests/test_redis.py (hunk 1 above)
Comment (shadchin) [reply instrumentation/opentelemetry-instrumentation-redis/tests/test_redis.py:292]: Sorry for late answer.

I looked at the code, connection parameters are not passed in this place

**Comment by xrmx** [comment]
@shadchin care to rebase and add a changelog entry please?

**Pull request merged**